my/src/components/Card.vue

```diff
@@ -174,35 +172,28 @@ export default Vue.extend({
       return '';
     },
     onDate(type: string, value: any) {
-      const day = this.inputType === 'date' ? this.dates[2].value : 1;
+      let year;
+      let month;
+      let day = this.inputType === 'date' ? this.dates[2].value : 1;
+
       if (type.toLowerCase() === 'year') {
-        this.$emit(
-          'input',
-          new Date(
-            value,
-            months.indexOf(this.dates[1].value),
-            day,
-          ),
-        );
+        year = value;
+        month = months.indexOf(this.dates[1].value);
       } else if (type.toLowerCase() === 'month') {
-        this.$emit(
-          'input',
-          new Date(
-            this.dates[0].value,
-            months.indexOf(value),
-            day,
-          ),
-        );
+        year = this.dates[0].value;
+        month = months.indexOf(value);
       } else {
-        this.$emit(
-          'input',
-          new Date(
-            this.dates[0].value,
-            months.indexOf(this.dates[1].value),
-            value,
-          ),
-        );
+        year = this.dates[0].value;
+        month = months.indexOf(this.dates[1].value);
+        day = value;
       }
+
+      this.$emit(
+        'input',
+        // When passed into Firebase function, Date objects need to be serialized or else they appear as empty objects
+        new Date(year, month, day).toJSON(),
+      );
```

Comment:
General refactoring and emitting a JSON object to the back-end, since normal Date objects get converted to empty objects (wtf Google??).

@amirafzali We talked about this before, but I actually think converting this to JSON is the play instead of using new Date().getTime() since new Date().toJson() makes it human-readable. If you're amiable to it, I'll update the back-end function with it.

Reply:
As long as Date().toJson() can be parsed by new Date() on all browser just as well as Date.getTime() can, I don't see why not. We'd have to update all references again but that's not a problem.